public/app/features/dimensions/editors/ResourcePickerPopover.tsx

```diff
@@ -83,7 +96,10 @@ export const ResourcePickerPopover = (props: Props) => {
               <Button
                 className={styles.button}
                 variant={newValue && newValue !== value ? 'primary' : 'secondary'}
-                onClick={() => onChange(newValue)}
+                onClick={() => {
+                  // TODO: if the trigger is from the upload tab, call backend /upload
```

Comment:
still TODO?  or did the upload already happen?

Reply:
still TODO, upload already happened before user hits Select. Currently /upload is called on the drag and drop action.